lib/dialects/sqlite/connection-manager.js

```diff
@@ -30,16 +31,25 @@ ConnectionManager.prototype.getConnection = function(options) {
   options.uuid = options.uuid || 'default';
   options.inMemory = ((self.sequelize.options.storage || self.sequelize.options.host || ':memory:') === ':memory:') ? 1 : 0;
 
-  if (self.connections[options.inMemory || options.uuid]) return Promise.resolve(self.connections[options.inMemory || options.uuid]);
+  var dialectOptions = self.sequelize.options.dialectOptions;
+  options.readWriteMode = dialectOptions && dialectOptions.mode;
+
+  if (self.connections[options.inMemory || options.uuid]) {
+    return Promise.resolve(self.connections[options.inMemory || options.uuid]);
+  }
 
   return new Promise(function (resolve, reject) {
-    self.connections[options.inMemory || options.uuid] = new self.lib.Database(self.sequelize.options.storage || self.sequelize.options.host || ':memory:', function(err) {
-      if (err) {
-        if (err.code === 'SQLITE_CANTOPEN') return reject(new sequelizeErrors.ConnectionError(err));
-        return reject(new sequelizeErrors.ConnectionError(err));
+    self.connections[options.inMemory || options.uuid] = new self.lib.Database(
+      self.sequelize.options.storage || self.sequelize.options.host || ':memory:',
+      options.readWriteMode || (sqlite3.OPEN_READWRITE | sqlite3.OPEN_CREATE), // default mode
```

Comment:
You should be able to use `this.lib` here instead, so we don't have to require sqlite3 at the top of the file

The reason we do it this way is, that it allows people to specify another library, which is api compatible (for example spatialite), using  `dialectModulePath`